starter-core/src/main/java/io/micronaut/starter/feature/architecture/Arm.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.architecture;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.feature.aws.Cdk;
+import io.micronaut.starter.feature.function.awslambda.AwsLambda;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class Arm implements CpuArchitecture {
```

Comment:
Why they need to be features? Can't we just use the `Architecture` enum? Pretty much as we do with `JdkVersion` for example.

Reply:
> Why they need to be features? Can't we use the `Architecture` enum? Pretty much as we do with `JdkVersion`, for example.

We could do it without features and add it to Options as you suggest.
https://github.com/micronaut-projects/micronaut-starter/blob/3.7.x/starter-core/src/main/java/io/micronaut/starter/options/Options.java

However, currently, this selection only influences a cloud vendor service. I think it is more flexible and unobtrusive as a feature. 

I would suggest leaving it as a feature for now but no strong opinion about this. I see your point.

Reply:
Another advantage of using it as a feature is that we don't need to modify the front end to support it immediately.

Reply:
Ok, fair enough, but then please remove the enum. `CpuArchitecture` is already a one-of feature. Note that currently for 2 options we have:
- `enum Architecture`.
- `class CpuArchitecture`.
- `class Arm`.
- `class X86`.

IMHO it is unnecessary complex.

Reply:
I have removed the enum. Removing it has forced me to do several `instanceof` and add `getListOption` method.  I think the code was simpler with the enum.  🤷🏻‍♂️

https://github.com/micronaut-projects/micronaut-starter/pull/1466/commits/5c24c4d0ea604521a532e8e7f153be948fe654c4